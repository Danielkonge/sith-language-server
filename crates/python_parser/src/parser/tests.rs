mod context;
mod function;
mod invalid_assignments;
mod parser;
mod suite;
