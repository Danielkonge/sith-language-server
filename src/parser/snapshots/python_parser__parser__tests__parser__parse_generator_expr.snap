---
source: src/parser/tests.rs
expression: "parse(\"\n(i for i in list)\n(a async for i in iter)\n(b for c in d if x in w if y and yy if z)\n(a for b in c if d and e for f in j if k > h)\n(a for b in c if d and e async for f in j if k > h)\nf(x for i in l)\nf(a, x for i in l)\nf(a, x for i, j in l)\n\")"
---
ParsedFile {
    ast: Module {
        body: [
            Expression(
                ExpressionStmt {
                    value: Generator(
                        GeneratorExpr {
                            element: Id(
                                Identifier {
                                    id: "i",
                                    range: 2..3,
                                },
                            ),
                            generators: [
                                Comprehension {
                                    target: Id(
                                        Identifier {
                                            id: "i",
                                            range: 8..9,
                                        },
                                    ),
                                    iter: Id(
                                        Identifier {
                                            id: "list",
                                            range: 13..17,
                                        },
                                    ),
                                    ifs: [],
                                    is_async: false,
                                    range: 4..17,
                                },
                            ],
                            range: 1..18,
                        },
                    ),
                    range: 1..18,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Generator(
                        GeneratorExpr {
                            element: Id(
                                Identifier {
                                    id: "a",
                                    range: 20..21,
                                },
                            ),
                            generators: [
                                Comprehension {
                                    target: Id(
                                        Identifier {
                                            id: "i",
                                            range: 32..33,
                                        },
                                    ),
                                    iter: Id(
                                        Identifier {
                                            id: "iter",
                                            range: 37..41,
                                        },
                                    ),
                                    ifs: [],
                                    is_async: true,
                                    range: 22..41,
                                },
                            ],
                            range: 19..42,
                        },
                    ),
                    range: 19..42,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Generator(
                        GeneratorExpr {
                            element: Id(
                                Identifier {
                                    id: "b",
                                    range: 44..45,
                                },
                            ),
                            generators: [
                                Comprehension {
                                    target: Id(
                                        Identifier {
                                            id: "c",
                                            range: 50..51,
                                        },
                                    ),
                                    iter: Id(
                                        Identifier {
                                            id: "d",
                                            range: 55..56,
                                        },
                                    ),
                                    ifs: [
                                        Compare(
                                            CompareExpr {
                                                left: Id(
                                                    Identifier {
                                                        id: "x",
                                                        range: 60..61,
                                                    },
                                                ),
                                                ops: [
                                                    In,
                                                ],
                                                comparators: [
                                                    Id(
                                                        Identifier {
                                                            id: "w",
                                                            range: 65..66,
                                                        },
                                                    ),
                                                ],
                                                range: 60..66,
                                            },
                                        ),
                                        BoolOp(
                                            BoolOpExpr {
                                                op: And,
                                                values: [
                                                    Id(
                                                        Identifier {
                                                            id: "y",
                                                            range: 70..71,
                                                        },
                                                    ),
                                                    Id(
                                                        Identifier {
                                                            id: "yy",
                                                            range: 76..78,
                                                        },
                                                    ),
                                                ],
                                                range: 70..78,
                                            },
                                        ),
                                        Id(
                                            Identifier {
                                                id: "z",
                                                range: 82..83,
                                            },
                                        ),
                                    ],
                                    is_async: false,
                                    range: 46..83,
                                },
                            ],
                            range: 43..84,
                        },
                    ),
                    range: 43..84,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Generator(
                        GeneratorExpr {
                            element: Id(
                                Identifier {
                                    id: "a",
                                    range: 86..87,
                                },
                            ),
                            generators: [
                                Comprehension {
                                    target: Id(
                                        Identifier {
                                            id: "b",
                                            range: 92..93,
                                        },
                                    ),
                                    iter: Id(
                                        Identifier {
                                            id: "c",
                                            range: 97..98,
                                        },
                                    ),
                                    ifs: [
                                        BoolOp(
                                            BoolOpExpr {
                                                op: And,
                                                values: [
                                                    Id(
                                                        Identifier {
                                                            id: "d",
                                                            range: 102..103,
                                                        },
                                                    ),
                                                    Id(
                                                        Identifier {
                                                            id: "e",
                                                            range: 108..109,
                                                        },
                                                    ),
                                                ],
                                                range: 102..109,
                                            },
                                        ),
                                    ],
                                    is_async: false,
                                    range: 88..109,
                                },
                                Comprehension {
                                    target: Id(
                                        Identifier {
                                            id: "f",
                                            range: 114..115,
                                        },
                                    ),
                                    iter: Id(
                                        Identifier {
                                            id: "j",
                                            range: 119..120,
                                        },
                                    ),
                                    ifs: [
                                        Compare(
                                            CompareExpr {
                                                left: Id(
                                                    Identifier {
                                                        id: "k",
                                                        range: 124..125,
                                                    },
                                                ),
                                                ops: [
                                                    Gt,
                                                ],
                                                comparators: [
                                                    Id(
                                                        Identifier {
                                                            id: "h",
                                                            range: 128..129,
                                                        },
                                                    ),
                                                ],
                                                range: 124..129,
                                            },
                                        ),
                                    ],
                                    is_async: false,
                                    range: 110..129,
                                },
                            ],
                            range: 85..130,
                        },
                    ),
                    range: 85..130,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Generator(
                        GeneratorExpr {
                            element: Id(
                                Identifier {
                                    id: "a",
                                    range: 132..133,
                                },
                            ),
                            generators: [
                                Comprehension {
                                    target: Id(
                                        Identifier {
                                            id: "b",
                                            range: 138..139,
                                        },
                                    ),
                                    iter: Id(
                                        Identifier {
                                            id: "c",
                                            range: 143..144,
                                        },
                                    ),
                                    ifs: [
                                        BoolOp(
                                            BoolOpExpr {
                                                op: And,
                                                values: [
                                                    Id(
                                                        Identifier {
                                                            id: "d",
                                                            range: 148..149,
                                                        },
                                                    ),
                                                    Id(
                                                        Identifier {
                                                            id: "e",
                                                            range: 154..155,
                                                        },
                                                    ),
                                                ],
                                                range: 148..155,
                                            },
                                        ),
                                    ],
                                    is_async: false,
                                    range: 134..155,
                                },
                                Comprehension {
                                    target: Id(
                                        Identifier {
                                            id: "f",
                                            range: 166..167,
                                        },
                                    ),
                                    iter: Id(
                                        Identifier {
                                            id: "j",
                                            range: 171..172,
                                        },
                                    ),
                                    ifs: [
                                        Compare(
                                            CompareExpr {
                                                left: Id(
                                                    Identifier {
                                                        id: "k",
                                                        range: 176..177,
                                                    },
                                                ),
                                                ops: [
                                                    Gt,
                                                ],
                                                comparators: [
                                                    Id(
                                                        Identifier {
                                                            id: "h",
                                                            range: 180..181,
                                                        },
                                                    ),
                                                ],
                                                range: 176..181,
                                            },
                                        ),
                                    ],
                                    is_async: true,
                                    range: 156..181,
                                },
                            ],
                            range: 131..182,
                        },
                    ),
                    range: 131..182,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Call(
                        CallExpr {
                            lhs: Id(
                                Identifier {
                                    id: "f",
                                    range: 183..184,
                                },
                            ),
                            args: Arguments {
                                args: [
                                    Generator(
                                        GeneratorExpr {
                                            element: Id(
                                                Identifier {
                                                    id: "x",
                                                    range: 185..186,
                                                },
                                            ),
                                            generators: [
                                                Comprehension {
                                                    target: Id(
                                                        Identifier {
                                                            id: "i",
                                                            range: 191..192,
                                                        },
                                                    ),
                                                    iter: Id(
                                                        Identifier {
                                                            id: "l",
                                                            range: 196..197,
                                                        },
                                                    ),
                                                    ifs: [],
                                                    is_async: false,
                                                    range: 187..197,
                                                },
                                            ],
                                            range: 185..197,
                                        },
                                    ),
                                ],
                                kw_args: [],
                                range: 184..198,
                            },
                            range: 183..198,
                        },
                    ),
                    range: 183..198,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Call(
                        CallExpr {
                            lhs: Id(
                                Identifier {
                                    id: "f",
                                    range: 199..200,
                                },
                            ),
                            args: Arguments {
                                args: [
                                    Id(
                                        Identifier {
                                            id: "a",
                                            range: 201..202,
                                        },
                                    ),
                                    Generator(
                                        GeneratorExpr {
                                            element: Id(
                                                Identifier {
                                                    id: "x",
                                                    range: 204..205,
                                                },
                                            ),
                                            generators: [
                                                Comprehension {
                                                    target: Id(
                                                        Identifier {
                                                            id: "i",
                                                            range: 210..211,
                                                        },
                                                    ),
                                                    iter: Id(
                                                        Identifier {
                                                            id: "l",
                                                            range: 215..216,
                                                        },
                                                    ),
                                                    ifs: [],
                                                    is_async: false,
                                                    range: 206..216,
                                                },
                                            ],
                                            range: 204..216,
                                        },
                                    ),
                                ],
                                kw_args: [],
                                range: 200..217,
                            },
                            range: 199..217,
                        },
                    ),
                    range: 199..217,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Call(
                        CallExpr {
                            lhs: Id(
                                Identifier {
                                    id: "f",
                                    range: 218..219,
                                },
                            ),
                            args: Arguments {
                                args: [
                                    Id(
                                        Identifier {
                                            id: "a",
                                            range: 220..221,
                                        },
                                    ),
                                    Generator(
                                        GeneratorExpr {
                                            element: Id(
                                                Identifier {
                                                    id: "x",
                                                    range: 223..224,
                                                },
                                            ),
                                            generators: [
                                                Comprehension {
                                                    target: Tuple(
                                                        TupleExpr {
                                                            elements: [
                                                                Id(
                                                                    Identifier {
                                                                        id: "i",
                                                                        range: 229..230,
                                                                    },
                                                                ),
                                                                Id(
                                                                    Identifier {
                                                                        id: "j",
                                                                        range: 232..233,
                                                                    },
                                                                ),
                                                            ],
                                                            range: 229..234,
                                                        },
                                                    ),
                                                    iter: Id(
                                                        Identifier {
                                                            id: "l",
                                                            range: 237..238,
                                                        },
                                                    ),
                                                    ifs: [],
                                                    is_async: false,
                                                    range: 225..238,
                                                },
                                            ],
                                            range: 223..238,
                                        },
                                    ),
                                ],
                                kw_args: [],
                                range: 219..239,
                            },
                            range: 218..239,
                        },
                    ),
                    range: 218..239,
                },
            ),
        ],
        range: 0..239,
    },
    parse_errors: [],
}
